Sources/Vehicle/Templates/CarPlayEntitiesListTemplate.swift

```diff
@@ -5,65 +5,112 @@ import PromiseKit
 import Shared
 
 @available(iOS 16.0, *)
-final class EntitiesListTemplate: CarPlayTemplateProvider {
-    enum GridPage {
-        case Next
-        case Previous
+final class CarPlayEntitiesListTemplate: CarPlayTemplateProvider {
+    enum FilterType {
+        case domain(String)
+        case areaId(entityIds: [String])
     }
 
     enum CPEntityError: Error {
         case unknown
     }
 
-    private let entityIconSize: CGSize = .init(width: 64, height: 64)
-    private let domain: String
+    private let filterType: FilterType
     private var server: Server
     private let entitiesCachedStates: HACache<HACachedStates>
     private var currentPage: Int = 0
 
     private var itemsPerPage: Int = CPListTemplate.maximumItemCount
     private var entitiesSubscriptionToken: HACancellable?
 
-    var template: CPTemplate
+    var template: CPListTemplate
     weak var interfaceController: CPInterfaceController?
 
-    init(title: String, domain: String, server: Server, entitiesCachedStates: HACache<HACachedStates>) {
-        self.domain = domain
+    private let entityIconSize: CGSize = .init(width: 64, height: 64)
+    private let entityIdKey = "entityId"
+
+    private let paginatedListTemplate: CarPlayPaginatedListTemplate
+    private var entitiesIdsCurrentlyInList: [String] = []
+
+    init(title: String, filterType: FilterType, server: Server, entitiesCachedStates: HACache<HACachedStates>) {
+        self.filterType = filterType
         self.server = server
         self.entitiesCachedStates = entitiesCachedStates
-        self.template = CPListTemplate(title: title, sections: [])
+        self.paginatedListTemplate = CarPlayPaginatedListTemplate(title: title, items: [])
+        self.template = paginatedListTemplate.template
     }
 
-    public func getTemplate() -> CPTemplate {
-        defer {
-            updateListItems()
+    func templateWillDisappear(template: CPTemplate) {
+        if self.template == template {
+            entitiesSubscriptionToken?.cancel()
+        }
+    }
+
+    func templateWillAppear(template: CPTemplate) {
+        if self.template == template {
+            update()
             entitiesSubscriptionToken = entitiesCachedStates.subscribe { [weak self] _, _ in
-                self?.updateListItems()
+                self?.update()
             }
         }
-
-        return template
     }
 
-    private func updateListItems() {
-        guard let entities = entitiesCachedStates.value, let listTemplate = template as? CPListTemplate else { return }
+    func update() {
+        guard let entities = entitiesCachedStates.value else { return }
```

Comment:
most all of this logic can be moved into a `map` of the `entitesCachedStates` turning this into a `HACache<[HAEntity]>`